src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -317,32 +316,44 @@ public Builder addAnnotation(AnnotationSpec annotationSpec) {
       return this;
     }
 
-    public Builder addAnnotation(Type annotation) {
+    public Builder addAnnotation(TypeName annotation) {
       return addAnnotation(AnnotationSpec.builder(annotation).build());
     }
 
+    public Builder addAnnotation(Type annotation) {
```

Comment:
Another.

Do we ever want to accept `Type`? Or do we just accept `Class<?>` and make you use a `TypeName` if you want something complex. I think that might have been what @gk5885 was saying today.